src/Annotation/README.md

```diff
@@ -0,0 +1,236 @@
+# Annotation feature
+
+This feature provides a system to declare annotations in docComment and 
+retrieve it into php classes.
+
+## Declaration 
+_Rules to be observed when declaring an annotation in the docComment_
+##### Identification
+The identification character of an annotation **MUST** be an `@`  
+e.g. `@Route`
+
+##### Annotation name
+The annotation name:
+- **MUST** start width uppercase. \[A-Z\].
+- Can only use alpha-numeric characters. [a-zA-Z]. It can't use special
+  characters or white spaces.   
+  
+e.g. `@MyAnnotationName`  
+_This option can be changed by changing the default annotation name declaration regex_
+
+  
+##### Options
+###### Declaration
+When options are needed it **MUST** be declared between parentheses.  
+e.g. `@Route(name="my_route")`  
+
+###### Separator
+Each option must be separated by a comma.  
+`option1, option2`
+
+###### Options types
+Four types can be declared in an annotation:
+<table>
+    <thead>
+        <tr>
+            <th>Type</th>
+            <th>Note</th>
+            <th>Example</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>String</td>
+            <td>Must be declared between double quotes</td>
+            <td>"my string"</td>
+        </tr>
+        <tr>
+            <td>Int</td>
+            <td>Can be declared without double quotes</td>
+            <td>1548</td>
+        </tr>
+        <tr>
+            <td>Bool</td>
+            <td>Declared width keywords <b>true</b> or <b>false</b></td>
+            <td>store = false</td>
+        </tr>
+        <tr>
+            <td>Array</td>
+            <td>
+                Must be daclared between braces.<br>
+                <i>Arrays can be nested to infinity</i>
+            </td>
+            <td>
+                { 1254, "Hello world" }<br>
+                { "hello world", { 1254, "test" }, 1845}
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+###### Key
+A key can be specified for any type of data, but it is optional.  
+In this case, the affectation operator `=` must be used.  
+
+e.g. `key = "my value"` or `myArray = { 1254, number = 545, "Hello world" }`  
+
+###### Level
+annotations can be declared at the class, method and attribute level.  
+
+## Annotation classes 
+Foreach annotation you **MUST** declare an annotation class who implements
+**AnnotationInterface**.  
+
+###### Attributes
```

Comment:
Codacy found an issue: [[heading-increment] Heading levels should increment by one level at a time](https://app.codacy.com/gh/pierregaimard/framework3/pullRequest?prid=6283010)